src/components/AddExperience/index.tsx

```diff
@@ -0,0 +1,273 @@
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import { CalendarIcon } from "lucide-react";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { format } from "date-fns";
+import { Calendar } from "@/components/ui/calendar";
+import { useEffect } from "react";
+import { employmentTypes } from "@/sample-data";
+import type { JobExperienceType } from "@/types";
+
+function AddExperience({
+  onSubmit,
+  onCancel,
+}: {
+  onSubmit: (newExperience: JobExperienceType) => void;
+  onCancel: () => void;
+}) {
+  const formSchema = z
```

Comment:
Consider declaring the schema outside of the component - right now you're recreating it on every render.